terraform/account-wide-infrastructure/modules/lambda-errors-metric-alarm/iam.tf

```diff
@@ -13,9 +13,43 @@ resource "aws_iam_policy" "lambda-errors-topic-kms-read-write" {
         ]
         Effect = "Allow"
         Resource = [
-          aws_kms_key.lambda-errors-topic-key.arn
+          aws_kms_key.lambda-errors-topic-key.arn,
+          aws_cloudwatch_metric_alarm.metric_alarm.arn
         ]
       }
     ]
   })
 }
+
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "sns_kms_key_policy" {
+  policy_id = "CloudWatchEncryptUsingKey"
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "kms:*"
+    ]
+    resources = ["*"]
```

Comment:
Is there a way to make this more specific?

Reply:
This enables IAM user permissions for the key so we kinda need it to be able to do stuff with this key, idk if making it more specific to sns and cloudwatch only work work because we need it to work with cloudwatch as a service as oppose to just that alarm